code/datums/components/nanites.dm

```diff
@@ -183,13 +185,67 @@
 	adjust_nanites(null, -amount)
 	return (nanite_volume > 0)
 
+///Modifies the current nanite volume, then checks if the nanites are depleted or exceeding the maximum amount
 /datum/component/nanites/proc/adjust_nanites(datum/source, amount)
 	SIGNAL_HANDLER
 
-	nanite_volume = clamp(nanite_volume + amount, 0, max_nanites)
+	nanite_volume += amount
+	if(nanite_volume > max_nanites)
+		reject_excess_nanites()
 	if(nanite_volume <= 0) //oops we ran out
 		qdel(src)
 
+/**
+  *	Handles how nanites leave the host's body if they find out that they're currently exceeding the maximum supported amount
+  *
+  * IC explanation:
+  * Normally nanites simply discard excess volume by slowing replication or 'sweating' it out in imperceptible amounts,
+  * but if there is a large excess volume, likely due to a programming change that leaves them unable to support their current volume,
+  * the nanites attempt to leave the host as fast as necessary to prevent nanite poisoning. This can range from minor oozing to nanites
+  * rapidly bursting out from every possible pathway, causing temporary inconvenience to the host.
+  */
+/datum/component/nanites/proc/reject_excess_nanites()
+	var/excess = nanite_volume - max_nanites
+	nanite_volume = max_nanites
+
+	switch(excess)
+		if(0 to 25) //Minor excess amount, the extra nanites are quietly expelled without visible effects
+			return
+		if(25.1 to 100) //Enough nanites getting rejected at once to be visible to the naked eye
+			host_mob.visible_message("<span class='warning'>A grainy grey slurry starts oozing out of [host_mob].</span>", "<span class='warning'>A grainy grey slurry starts oozing out of your skin.</span>", null, 4);
+		if(100.1 to 350) //Nanites getting rejected in massive amounts, but still enough to make a semi-orderly exit through vomit
+			if(iscarbon(host_mob))
+				var/mob/living/carbon/C = host_mob
+				host_mob.visible_message("<span class='warning'>[host_mob] vomits a grainy grey slurry!</span>", "<span class='warning'>You suddenly vomit a metallic-tasting grainy grey slurry!</span>", null);
+				C.vomit(0, FALSE, TRUE, FLOOR(excess / 100, 1), FALSE, VOMIT_NANITE, FALSE, TRUE, 0)
```

Comment:
Is this the sort that stuns/knocks down?

Reply:
Yes, stun is set to TRUE